### PR TITLE
add logging to Simulation Service

### PR DESCRIPTION
**Pull request by lezrak**
logging of crucial flow steps added

### CODE REVIEW

File: src/main/java/pl/neurocars/backend/simulation/SimulationServiceImpl.java [hunk 1]
```diff
@@ -72,7 +80,12 @@ public Simulation sendModelToAi(Simulation simulation) {
                 .block(Duration.ofSeconds(2));
 
         if (responseEntity != null && responseEntity.getStatusCodeValue() == 200) {
+            logger.info("sent model of simulation " + simulation.getId());
             simulation.setState(State.AWAITING_MODEL);
+        } else if (responseEntity != null) {
+            logger.info("failed to send model to AI for simulation " + simulation.getId() + "response code " + responseEntity.getStatusCodeValue());
```
Comment (BartoszF) [inline src/main/java/pl/neurocars/backend/simulation/SimulationServiceImpl.java]: should be error

File: src/main/java/pl/neurocars/backend/simulation/SimulationServiceImpl.java [hunk 2]
```diff
@@ -72,7 +80,12 @@ public Simulation sendModelToAi(Simulation simulation) {
...
+        } else {
+            logger.info("failed to send model to AI for simulation " + simulation.getId() + "no response");
```
Comment (BartoszF) [inline src/main/java/pl/neurocars/backend/simulation/SimulationServiceImpl.java]: should be error

File: src/main/java/pl/neurocars/backend/simulation/SimulationServiceImpl.java (hunk 1 above)
Comment (lezrak) [reply src/main/java/pl/neurocars/backend/simulation/SimulationServiceImpl.java]: fixed

File: src/main/java/pl/neurocars/backend/simulation/SimulationServiceImpl.java (hunk 2 above)
Comment (lezrak) [reply src/main/java/pl/neurocars/backend/simulation/SimulationServiceImpl.java]: fixed

**Pull request merged**